### PR TITLE
disallow the log.V(i).Infof syntax

### PR DESCRIPTION
**Pull request by tbg**
if you use it, you'll call Infof directly from the glog package,
and not from our local log package. That hasn't been an issue
so far, but it will be soon so this puts an end to it.

We might have an API like that again later, but let's keep it
simple for now.

### CODE REVIEW

File: util/log/log.go [hunk 1]
```diff
@@ -99,4 +99,9 @@ var Fatalln = glog.Fatalln
 var FatalDepth = glog.FatalDepth
 
 // V wraps glog.V. See that documentation for details.
-var V = glog.V
```
Comment (tschottdorf) [inline util/log/log.go]: this is the only interesting diff in this PR.

**Comment by spencerkimball** [comment]
LGTM

**Comment by bdarnell** [comment]
This breaks `--vmodule` by moving the call to `V()` from the original source file to the `log` module (I for one depend on `--vmodule` for debugging raft). There is no `VDepth()` function (as there is for `InfoDepth`, etc), so we can't get the old functionality back and still disallow calls to `V(n).Infof`. We should probably investigate moving from glog to something more flexible and better-maintained.

**Comment by tschottdorf** [comment]
@bdarnell looking at `multiraft/raft.go` it seems like exactly the same should happen now compared to before. The only change is in not being able to chain `log.V(x).*` but I don't see where you do that there.

**Comment by tschottdorf** [comment]
Oh I see what you're saying. Yeah, I think we'll want to switch logging frameworks at some point.

**Comment by tschottdorf** [comment]
Ok maybe I don't see what you're saying. If you're only using the `if log.V(x) { ... }` syntax (as I think you are?) then everything should be the same.

**Comment by bdarnell** [comment]
The implementation of `glog.V()` uses runtime.Caller to get the file it was called from. Before, glog.V was called directly from multiraft.go (the alias `var V = glog.V` did not introduce a new stack frame). The new function in log.go introduces a stack frame, so now the caller that glog sees is always log.go.

**Comment by tschottdorf** [comment]
That's awkward indeed. We should switch away from glog. What do you suggest in the meantime? I can just revert the change in multiraft since we depend on those logs for troubleshooting. Or we can run our own copy of glog temporarily in our source tree and add VDepth.

**Comment by bdarnell** [comment]
For now I suggest reverting the change in log.go but leaving everything else alone. `V(n).Infof` will no longer be a compilation error, but we are no longer using it so it won't matter too much. We'll just have to be disciplined to avoid re-introducing this usage until the switch to the new scheme is complete.

**Pull request merged**